### PR TITLE
Allow workers to request API from the thread that spawned them.

### PR DESCRIPTION
**Pull request by VladaLeykin**
This PR adds new ability to pm-rpc:

Workers that were created by main thread may now request API from the thread that has created them.
For that to happen main thread need to pass instances of Workers to pm-rpc to create eventHandlers for request messages.

### CODE REVIEW

**Comment by VladaLeykin** [comment]
This is work in progress. I'm working on tests now

**Comment by VladaLeykin** [comment]
Many integration tests have been disabled in this PR. 
The reason for this is because they are flaky/failing.

I've found that `fdescribe` commited to `integration/deep-namespaces.spec.js` disabled all integration tests from running, but the reporter has reported success for those test, giving the illusion they are running and good.

I haven't seen the exact reason for failure, I'll open issue for further investigation of those failing/flaky tests

File: .eslintrc.json [hunk 1]
```diff
@@ -14,6 +14,13 @@
...
     "indent":"off",
-    "sort-imports": "off"
+    "sort-imports": "off",
+    "no-restricted-globals": [2, {
+      "name": "fdescribe",
+      "message": "Do not commit fdescribe."
+    }, {
+      "name": "fit",
+      "message": "Do not commit fit."
+    }]
```
Comment (maoryosef2) [inline .eslintrc.json:24]: prefer to use jasmine/no-focused-tests

File: integration/content/client-worker/worker.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+onmessage = function (event) {
+    if (event.data.message === 'init') {
+        pmrpc.api.request('api-for-worker')
+            .then(api => {
+                if (api.add) {
+                    api.add(2, 3).then(result => {
+                        postMessage({message: 'api received', result})
+                    })
+                }
```
Comment (maoryosef2) [inline integration/content/client-worker/worker.js:11]: if `api.add` does not exists, your test will hang until timeout,
please send `postMessage({message: 'test failed' })` if api is not correct
and in the test itself, if the message was `test failed` call `done.fail()`

File: .eslintrc.json (hunk 1 above)
Comment (VladaLeykin) [reply .eslintrc.json:24]: will do

**Pull request merged**